community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java

```diff
@@ -382,8 +390,14 @@ public static void main( String[] incomingArguments ) throws IOException
      */
     public static void main( String[] incomingArguments, boolean defaultSettingsSuitableForTests ) throws IOException
     {
-        System.err.println("WARNING: neo4j-import is deprecated and support for it will be removed in a future\n" +
-                "version of Neo4j; please use neo4j-admin import instead.\n");
+        System.err.println( format( "WARNING: neo4j-import is deprecated and support for it will be removed in a future%n" +
+                "version of Neo4j; please use neo4j-admin import instead." ) );
+        if ( Runtime.getRuntime().maxMemory() > RECOMMENDED_MAX_HEAP_SIZE )
+        {
+            System.err.println( format( "WARNING: heap size is unnecessarily big, potentially causing garbage collection issues and%n" +
```

Comment:
its not clear about what gc issues we are talking about here?

Reply:
Long GC pauses if running into full GC. Perhaps it could be framed differently, without mentioning vague GC issues.

Reply:
Maybe here we can warn instead about crossing compact heap representation boundary and a general recommendation to keep heap as small as possible if you not sure that you need that much?